<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Populations | ddRADSeq Timon</title>
  <meta name="description" content="This is a step by step workflow for filtering reads generated by ddRADSeq with Stacks." />
  <meta name="generator" content="bookdown 0.38 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Populations | ddRADSeq Timon" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://www.inaturalist.org/observations/165892004" />
  <meta property="og:description" content="This is a step by step workflow for filtering reads generated by ddRADSeq with Stacks." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Populations | ddRADSeq Timon" />
  
  <meta name="twitter:description" content="This is a step by step workflow for filtering reads generated by ddRADSeq with Stacks." />
  <meta name="twitter:image" content="https://www.inaturalist.org/observations/165892004" />

<meta name="author" content="Adam Marques" />


<meta name="date" content="2024-04-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="denovo-map.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="initialize.html"><a href="initialize.html"><i class="fa fa-check"></i><b>1</b> Initialize</a>
<ul>
<li class="chapter" data-level="1.1" data-path="initialize.html"><a href="initialize.html#initialize-directories"><i class="fa fa-check"></i><b>1.1</b> INITIALIZE DIRECTORIES</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="trimmomatic.html"><a href="trimmomatic.html"><i class="fa fa-check"></i><b>2</b> Trimmomatic</a>
<ul>
<li class="chapter" data-level="2.1" data-path="trimmomatic.html"><a href="trimmomatic.html#sample-prep"><i class="fa fa-check"></i><b>2.1</b> Sample Prep</a></li>
<li class="chapter" data-level="2.2" data-path="trimmomatic.html"><a href="trimmomatic.html#trimmomatic-1"><i class="fa fa-check"></i><b>2.2</b> Trimmomatic</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="process-radtags.html"><a href="process-radtags.html"><i class="fa fa-check"></i><b>3</b> Process Radtags</a></li>
<li class="chapter" data-level="4" data-path="denovo-map.html"><a href="denovo-map.html"><i class="fa fa-check"></i><b>4</b> Denovo Map</a></li>
<li class="chapter" data-level="5" data-path="populations.html"><a href="populations.html"><i class="fa fa-check"></i><b>5</b> Populations</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ddRADSeq Timon</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="populations" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Populations<a href="populations.html#populations" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The populations program will analyze a population of individual samples computing a number of population genetics statistics as well as exporting a variety of standard output formats.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="populations.html#cb42-1" tabindex="-1"></a><span class="do">##############################</span></span>
<span id="cb42-2"><a href="populations.html#cb42-2" tabindex="-1"></a><span class="do">## INITIALIZE DIRECTORIES</span></span>
<span id="cb42-3"><a href="populations.html#cb42-3" tabindex="-1"></a><span class="do">##############################</span></span>
<span id="cb42-4"><a href="populations.html#cb42-4" tabindex="-1"></a><span class="co"># Define directories</span></span>
<span id="cb42-5"><a href="populations.html#cb42-5" tabindex="-1"></a>home_dir <span class="ot">&lt;-</span> <span class="st">&quot;./&quot;</span></span>
<span id="cb42-6"><a href="populations.html#cb42-6" tabindex="-1"></a>raw_dir <span class="ot">&lt;-</span> <span class="st">&quot;./raw/&quot;</span></span>
<span id="cb42-7"><a href="populations.html#cb42-7" tabindex="-1"></a>int_dir <span class="ot">&lt;-</span> <span class="st">&quot;./intermediate/&quot;</span></span>
<span id="cb42-8"><a href="populations.html#cb42-8" tabindex="-1"></a><span class="co"># results directories</span></span>
<span id="cb42-9"><a href="populations.html#cb42-9" tabindex="-1"></a>radtag_dir <span class="ot">&lt;-</span> <span class="fu">paste0</span>(int_dir,<span class="st">&quot;process_radtags/&quot;</span>)</span>
<span id="cb42-10"><a href="populations.html#cb42-10" tabindex="-1"></a>denovo_dir <span class="ot">&lt;-</span> <span class="fu">paste0</span>(int_dir,<span class="st">&quot;denovomap/&quot;</span>)</span>
<span id="cb42-11"><a href="populations.html#cb42-11" tabindex="-1"></a>populations_dir <span class="ot">&lt;-</span> <span class="fu">paste0</span>(int_dir,<span class="st">&quot;populations/&quot;</span>)</span></code></pre></div>
<p>The populations program provides filtering options to only include loci or variant sites that occur at certain frequencies in each population or in the metapopulation.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="populations.html#cb43-1" tabindex="-1"></a><span class="do">##############################</span></span>
<span id="cb43-2"><a href="populations.html#cb43-2" tabindex="-1"></a><span class="do">## PARAMETERS</span></span>
<span id="cb43-3"><a href="populations.html#cb43-3" tabindex="-1"></a><span class="do">##############################</span></span>
<span id="cb43-4"><a href="populations.html#cb43-4" tabindex="-1"></a><span class="do">### load parameters</span></span>
<span id="cb43-5"><a href="populations.html#cb43-5" tabindex="-1"></a><span class="co"># load population map</span></span>
<span id="cb43-6"><a href="populations.html#cb43-6" tabindex="-1"></a>pop_map <span class="ot">=</span> <span class="fu">paste0</span>(raw_dir,<span class="st">&quot;popmap&quot;</span>)</span>
<span id="cb43-7"><a href="populations.html#cb43-7" tabindex="-1"></a>M <span class="ot">=</span> <span class="dv">4</span>       <span class="co"># max mismatches allowed between stacks within individuals</span></span>
<span id="cb43-8"><a href="populations.html#cb43-8" tabindex="-1"></a>Thread <span class="ot">=</span> <span class="dv">10</span> <span class="co"># the number of threads/CPUs to use (default: 1).</span></span>
<span id="cb43-9"><a href="populations.html#cb43-9" tabindex="-1"></a>msp <span class="ot">=</span> <span class="fl">0.75</span>  <span class="co"># min number of samples per population</span></span>
<span id="cb43-10"><a href="populations.html#cb43-10" tabindex="-1"></a>maf <span class="ot">=</span> <span class="fl">0.05</span>  <span class="co"># minimum minor allele frequency </span></span>
<span id="cb43-11"><a href="populations.html#cb43-11" tabindex="-1"></a>moh <span class="ot">=</span> <span class="fl">0.5</span>   <span class="co"># maximum observed heterozygosity</span></span></code></pre></div>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="populations.html#cb44-1" tabindex="-1"></a><span class="co"># new list of directories to save results</span></span>
<span id="cb44-2"><a href="populations.html#cb44-2" tabindex="-1"></a>denovo_M_dir <span class="ot">&lt;-</span> <span class="fu">paste0</span>(denovo_dir,<span class="st">&quot;denovo_M&quot;</span>,M,<span class="st">&quot;/&quot;</span>)</span>
<span id="cb44-3"><a href="populations.html#cb44-3" tabindex="-1"></a><span class="do">## Run populations</span></span>
<span id="cb44-4"><a href="populations.html#cb44-4" tabindex="-1"></a><span class="co"># run denovo_map</span></span>
<span id="cb44-5"><a href="populations.html#cb44-5" tabindex="-1"></a><span class="fu">system</span>(<span class="fu">paste0</span>(<span class="st">&quot;populations&quot;</span>,</span>
<span id="cb44-6"><a href="populations.html#cb44-6" tabindex="-1"></a>              <span class="st">&quot; -P &quot;</span>, denovo_M_dir, </span>
<span id="cb44-7"><a href="populations.html#cb44-7" tabindex="-1"></a>              <span class="st">&quot; -O &quot;</span>, populations_dir,</span>
<span id="cb44-8"><a href="populations.html#cb44-8" tabindex="-1"></a>              <span class="st">&quot; -M &quot;</span>, pop_map,</span>
<span id="cb44-9"><a href="populations.html#cb44-9" tabindex="-1"></a>              <span class="st">&quot; --genepop --structure --fasta-loci&quot;</span>,</span>
<span id="cb44-10"><a href="populations.html#cb44-10" tabindex="-1"></a>              <span class="st">&quot; -t &quot;</span>,Thread,</span>
<span id="cb44-11"><a href="populations.html#cb44-11" tabindex="-1"></a>              <span class="st">&quot; --min-samples-per-pop &quot;</span>, <span class="fl">0.75</span>,</span>
<span id="cb44-12"><a href="populations.html#cb44-12" tabindex="-1"></a>              <span class="st">&quot; --min-maf &quot;</span>,<span class="fl">0.05</span>,</span>
<span id="cb44-13"><a href="populations.html#cb44-13" tabindex="-1"></a>              <span class="st">&quot;--max-obs-het&quot;</span>,<span class="fl">0.5</span>), </span>
<span id="cb44-14"><a href="populations.html#cb44-14" tabindex="-1"></a>       <span class="at">intern =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Warning in system(paste0(&quot;populations&quot;, &quot; -P &quot;, denovo_M_dir, &quot; -O &quot;, populations_dir, : running command
## &#39;populations -P ./intermediate/denovomap/denovo_M4/ -O ./intermediate/populations/ -M ./raw/popmap --genepop
## --structure --fasta-loci -t 10 --min-samples-per-pop 0.75 --min-maf 0.05--max-obs-het0.5&#39; had status 1</code></pre>
<pre><code>##  [1] &quot;Logging to &#39;./intermediate/populations/populations.log&#39;.&quot;                                                                           
##  [2] &quot;Locus/sample distributions will be written to &#39;./intermediate/populations/populations.log.distribs&#39;.&quot;                               
##  [3] &quot;populations parameters selected:&quot;                                                                                                   
##  [4] &quot;  Percent samples limit per population: 0.75&quot;                                                                                       
##  [5] &quot;  Locus Population limit: 1&quot;                                                                                                        
##  [6] &quot;  Percent samples overall: 0&quot;                                                                                                       
##  [7] &quot;  Minor allele frequency cutoff: 0.05&quot;                                                                                              
##  [8] &quot;  Maximum observed heterozygosity cutoff: 1&quot;                                                                                        
##  [9] &quot;  Applying Fst correction: none.&quot;                                                                                                   
## [10] &quot;  Pi/Fis kernel smoothing: off&quot;                                                                                                     
## [11] &quot;  F-stats kernel smoothing: off&quot;                                                                                                    
## [12] &quot;  Bootstrap resampling: off&quot;                                                                                                        
## [13] &quot;&quot;                                                                                                                                   
## [14] &quot;Parsing population map...&quot;                                                                                                          
## [15] &quot;The population map contained 3 samples, 1 population(s), 1 group(s).&quot;                                                               
## [16] &quot;Working on 3 samples.&quot;                                                                                                              
## [17] &quot;Working on 1 population(s):&quot;                                                                                                        
## [18] &quot;    Timon: GVA1418, GVA1515, GVA1926&quot;                                                                                               
## [19] &quot;Working on 1 group(s) of populations:&quot;                                                                                              
## [20] &quot;    defaultgrp: Timon&quot;                                                                                                              
## [21] &quot;&quot;                                                                                                                                   
## [22] &quot;Polymorphic sites in GenePop format will be written to &#39;./intermediate/populations/populations.snps.genepop&#39;&quot;                       
## [23] &quot;Polymorphic loci in GenePop format will be written to &#39;./intermediate/populations/populations.haps.genepop&#39;&quot;                        
## [24] &quot;Polymorphic sites in Structure format will be written to &#39;./intermediate/populations/populations.structure&#39;&quot;                        
## [25] &quot;FASTA consensus sequences for each locus in the metapopulation  will be written to &#39;./intermediate/populations/populations.loci.fa&#39;&quot;
## [26] &quot;Raw haplotypes will be written to &#39;./intermediate/populations/populations.haplotypes.tsv&#39;&quot;                                          
## [27] &quot;Population-level summary statistics will be written to &#39;./intermediate/populations/populations.sumstats.tsv&#39;&quot;                       
## [28] &quot;Population-level haplotype summary statistics will be written to &#39;./intermediate/populations/populations.hapstats.tsv&#39;&quot;             
## [29] &quot;&quot;                                                                                                                                   
## [30] &quot;Processing data in batches:&quot;                                                                                                        
## [31] &quot;  * load a batch of catalog loci and apply filters&quot;                                                                                 
## [32] &quot;  * compute SNP- and haplotype-wise per-population statistics&quot;                                                                      
## [33] &quot;  * write the above statistics in the output files&quot;                                                                                 
## [34] &quot;  * export the genotypes/haplotypes in specified format(s)&quot;                                                                         
## [35] &quot;More details in &#39;./intermediate/populations/populations.log.distribs&#39;.&quot;                                                             
## [36] &quot;&quot;                                                                                                                                   
## [37] &quot;Now processing...&quot;                                                                                                                  
## [38] &quot;Batch 1 &quot;                                                                                                                           
## [39] &quot;&quot;                                                                                                                                   
## [40] &quot;Removed 197 loci that did not pass sample/population constraints from 198 loci.&quot;                                                    
## [41] &quot;Kept 1 loci, composed of 148 sites; 0 of those sites were filtered, 0 variant sites remained.&quot;                                      
## [42] &quot;Number of loci with PE contig: 1.00 (100.0%);&quot;                                                                                      
## [43] &quot;  Mean length of loci: 138.00bp (stderr inf);&quot;                                                                                      
## [44] &quot;Number of loci with SE/PE overlap: 1.00 (100.0%);&quot;                                                                                  
## [45] &quot;  Mean length of overlapping loci: 148.00bp (stderr -nan); mean overlap: 31.00bp (stderr -nan);&quot;                                    
## [46] &quot;Mean genotyped sites per locus: 148.00bp (stderr -nan).&quot;                                                                            
## [47] &quot;&quot;                                                                                                                                   
## [48] &quot;Population summary statistics (more detail in populations.sumstats_summary.tsv):&quot;                                                   
## [49] &quot;  Timon: -nan samples per locus; pi: -nan; all/variant/polymorphic sites: 148/0/0; private alleles: 0&quot;                              
## attr(,&quot;status&quot;)
## [1] 1</code></pre>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="populations.html#cb47-1" tabindex="-1"></a><span class="co"># cleanup</span></span>
<span id="cb47-2"><a href="populations.html#cb47-2" tabindex="-1"></a><span class="fu">rm</span>(pop_map,M,Thread,msp,maf,moh,denovo_M_dir)</span></code></pre></div>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="denovo-map.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/05-populations.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
